tensornetwork/block_sparse/blocksparsetensor.py

```diff
@@ -391,6 +391,27 @@ def __rmul__(self, number: np.number) -> "ChargeArray":
   def __truediv__(self, number: np.number) -> "ChargeArray":
     raise NotImplementedError("__truediv__ not implemented for ChargeArray")
 
+  def __repr__(self):
+    if len(self._charges) > 0:
+      charge_types = self._charges[0].names
+    else:
+      charge_types = 'no charge types (scalar)'
+    output = 'BlockSparseTensor\n   shape: ' + repr(
+        self.shape
+    ) + '\n   charge types: ' + charge_types + '\n   dtype: ' + repr(
+        self.dtype.name) + '\n   flat flows: ' + repr(
+            self.flat_flows) + '\n   order: ' + repr(
+                self._order) + '\n   data:' + repr(self.data)
```

Comment:
Do you have an example of what this prints out?

Reply:
yes, looks like this:
```
BlockSparseTensor
   shape: (10, 10)
   charge types: ['U1Charge', 'Z2Charge']
   dtype: 'float64'
   flat flows: [True, False]
   order: [[0], [1]]
   data:array([0.36553923, 0.14096483, 0.74018218, 0.42214647, 0.0460793 ,
       0.03905852, 0.57348562, 0.47659091, 0.12239714, 0.01138399,
       0.47512114, 0.07621713, 0.02356647, 0.60349018])

```

Reply:
beautiful!

Reply:
thanks, I  put some effort in it ;)